gobblin-utility/src/test/java/gobblin/util/ExecutorsUtilsTest.java

```diff
@@ -47,4 +52,22 @@ public void run() {
 
     thread.run();
   }
+
+  @Test
+  public void testParallelize() {
+    List<Integer> nums = Lists.newArrayList();
+    nums.add(3);
+    nums.add(10);
+    final int factor = 5;
+
+    Function<Integer, String> multiply = new Function<Integer, String>() {
+      @Override
+      public String apply(Integer input) {
+        return Integer.toString(input * factor);
+      }
+    };
+
+    List<String> result = ExecutorsUtils.parallelize(nums, multiply , 2, 60, Optional.<Logger>absent());
+    Assert.assertEquals(Arrays.asList("15","50"), result);
+  }
 }
```

Comment:
We need tests for the following cases:
* A function call throws an Exception
* A function call times out